lib/src/components/supa_socials_auth.dart

```diff
@@ -147,75 +158,90 @@ class _SupaSocialsAuthState extends State<SupaSocialsAuth> {
-            height: 48,
+        Widget iconWidget = SizedBox(
+          height: 48,
+          width: 48,
+          child: Icon(socialProvider._iconData),
+        );
+
+        Color? foregroundColor = coloredBg ? Colors.white : null;
+        Color? backgroundColor = coloredBg ? socialProvider._btnBgColor : null;
+        Color? overlayColor = coloredBg ? Colors.white10 : null;
+
+        Color? iconColor = coloredBg ? Colors.white : null;
+
+        if (socialProvider == SocialProviders.google && coloredBg) {
+          iconWidget = Image.asset(
+            'assets/logos/google_light.png',
+            package: 'supabase_auth_ui',
             width: 48,
-            child: Icon(socialProvider._iconData),
+            height: 48,
           );
 
-          Color? foregroundColor = coloredBg ? Colors.white : null;
-          Color? backgroundColor =
-              coloredBg ? socialProvider._btnBgColor : null;
-          Color? overlayColor = coloredBg ? Colors.white10 : null;
-
-          Color? iconColor = coloredBg ? Colors.white : null;
+          foregroundColor = Colors.black;
+          backgroundColor = Colors.white;
+          overlayColor = Colors.white;
+        }
 
-          if (socialProvider == SocialProviders.google && coloredBg) {
-            iconWidget = Image.asset(
-              'assets/logos/google_light.png',
-              package: 'supabase_auth_ui',
-              width: 48,
-              height: 48,
+        onAuthButtonPressed() async {
+          try {
+            await supaClient.auth.signInWithOAuth(
+              socialProvider.provider,
+              redirectTo: widget.redirectUrl,
             );
-
-            foregroundColor = Colors.black;
-            backgroundColor = Colors.white;
-            overlayColor = Colors.white;
+          } on AuthException catch (error) {
+            if (widget.onError == null) {
+              context.showErrorSnackBar(error.message);
+            } else {
+              widget.onError?.call(error);
+            }
+          } catch (error) {
+            if (widget.onError == null) {
+              context.showErrorSnackBar('Unexpected error has occurred: $error');
+            } else {
+              widget.onError?.call(error);
+            }
           }
+        }
+
+        final authButtonStyle = ButtonStyle(
+          foregroundColor: MaterialStateProperty.all(foregroundColor),
+          backgroundColor: MaterialStateProperty.all(backgroundColor),
+          overlayColor: MaterialStateProperty.all(overlayColor),
+          iconColor: MaterialStateProperty.all(iconColor),
+        );
+
+        return Padding(
+          padding: const EdgeInsets.symmetric(vertical: 8.0),
+          child: widget.socialButtonVariant == SocialButtonVariant.icon
+              ? IconButton(
+                  icon: iconWidget,
+                  style: authButtonStyle,
+                  onPressed: onAuthButtonPressed,
+                )
```

Comment:
Would it make sense to maybe use an ElevatedButton here instead of IconButton? That way, we can keep each icon's background color.

Reply:
If we use an elevated button it looks like this, which is not ideal.
<img width="1424" alt="image" src="https://user-images.githubusercontent.com/69860033/223434760-2c9a1066-41d4-4276-811f-fc02e428fe63.png">

Reply:
I change it to use a circle avatar wrapping the icon button. This is the result
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/69860033/223436230-70ec01fe-1826-408e-9910-385af488938e.png">